### PR TITLE
Add jax.numpy.signbit

### PR DESCRIPTION
**Pull request by dtran24**
Opened this as per the reviewer's [suggestion](https://github.com/ekzhang/jax-js/pull/160#pullrequestreview-4840340123). Can close https://github.com/ekzhang/jax-js/pull/160 if this PR gets merged

## Impact

Users can detect negative values, negative zero, and signed NaNs where the backend preserves the NaN sign. Integer and boolean inputs follow NumPy/JAX semantics and return boolean arrays.

## Validation

- `pnpm build`
- `pnpm check`
- `pnpm lint --max-warnings 0`
- `pnpm format:check`
- targeted signbit and dtype suites on Chromium, Firefox, and WebKit
- full Chromium suite: 2,487 passed; two existing WebGPU FFT tolerance failures reproduced on upstream `main`

### CODE REVIEW

File: src/library/numpy.ts [hunk 1]
```diff
@@ -1899,6 +1899,30 @@ export function sign(x: ArrayLike): Array {
   return where(notEqual(x.ref, 0), where(less(x, 0), -1, 1), 0);
 }
 
+/**
+ * Test element-wise whether the sign bit is set.
+ *
+ * Unlike comparisons with zero, this distinguishes `-0` from `0`. The sign of
+ * NaN is also preserved, except where the platform canonicalizes NaN: JS
+ * engines may do so when the cpu backend reads floats out of typed arrays, and
+ * conversions of f16 NaN on GPU backends may lose the sign as well.
```
Comment (dtran24) [inline src/library/numpy.ts:1908]: Thought this was fine since the repo has precedent of some devices having edge cases already
https://github.com/ekzhang/jax-js/blob/e0d09c72d6b5a9279c6f38d9a6b7c9c25958a247/test/setup.ts#L33-L39

Can open a new issue to tackle this in future if we'd like

File: test/setup.ts [hunk 2]
```diff
@@ -37,3 +37,12 @@ expect.extend({
...
   return device !== "webgl";
 }
+
+/**
+ * The cpu backend reads floats out of typed arrays as JS numbers, and some JS
+ * engines canonicalize NaN in the process, losing its sign. We skip tests that
+ * depend on the sign of NaN for these devices.
+ */
+export function preservesNanSign(device: Device): boolean {
+  return device !== "cpu" && hasStrictNumerics(device);
```
Comment (ekzhang) [inline test/setup.ts]: Hmm I'd prefer if we just had one hasStrictNumerics flag and not multiple based on specific numeric features, if possible - webgl is mostly less supported because it's not worth it

File: src/library/numpy.ts (hunk 1 above)
Comment (ekzhang) [reply src/library/numpy.ts:1908]: Hmm yeah to be honest nan is not sanely handled in any backend right now because the presence of nan is technically UB in webgpu and webgl.

jax-js's approach is to try our best but not guarantee anything so this is fine

**Review by ekzhang — APPROVED** [review]
lgtm can you remove the extra test flag?

**Comment by ekzhang** [comment]
> existing WebGPU FFT tolerance failures

also sorry about this, hmm it passes in CI, do you know which tests these are we need to fix?

File: test/setup.ts (hunk 2 above)
Comment (dtran24) [reply test/setup.ts]: Removed in https://github.com/ekzhang/jax-js/pull/161/commits/ebd594f69aec692f317677550d2f37eb60597fdf

**Comment by dtran24** [comment]
@ekzhang Np! The two tests are
https://github.com/ekzhang/jax-js/blob/a0e12606d76d51e0da7784fb968bcca43cf0e5fd/test/numpy-fft.test.ts#L63-L82
https://github.com/ekzhang/jax-js/blob/a0e12606d76d51e0da7784fb968bcca43cf0e5fd/test/numpy-fft.test.ts#L123-L140

These fail for me only on WebGPU. The expected imaginary values are zero, but Chromium on my Apple M5 Max returns small residuals around 2.6e-7 to 7.0e-7. That exceeds allclose’s default `atol=1e-7`

I reproduced the same failures on upstream main, so they are unrelated to this PR. Ubuntu CI skips WebGPU; macOS CI runs it and passes. The likely difference is the GPU adapter or driver since WebGPU floating-point rounding can vary slightly between implementations

A small follow-up fix so the tests can pass on this setup could be adding `{ atol: 1e-6 }` to the two `result.imag` assertions. Happy to look into making this change
